Makefile

```diff
@@ -1,40 +1,50 @@
-all: deps
+UNAME_S := $(shell uname -s)
 
-deps: get-submodules c4 z3
+ifeq ($(UNAME_S), Darwin)
+	LIB_C4 = lib/c4/build/src/libc4/libc4.dylib
+	LIB_Z3 = lib/z3/build/libz3.dylib
+else
+	LIB_C4 = lib/c4/build/src/libc4/libc4.so
+	LIB_Z3 = lib/z3/build/libz3.so
+endif
 
-clean-deps:
-	rm -r lib/c4/build
-	rm -r lib/z3/build
+LIBRARY_PATH = $(shell pwd)/$(shell dirname ${LIB_C4}):$(shell pwd)/$(shell dirname ${LIB_Z3})
+LD_LIBRARY_PATH = ${LIBRARY_PATH}:$(shell $$LD_LIBRARY_PATH)
+DYLD_LIBRARY_PATH = ${LIBRARY_PATH}:$(shell $$DYLD_LIBRARY_PATH)
+
+all: deps
 
-c4: lib/c4/build/src/libc4/libc4.dylib
+deps: ${LIB_C4} ${LIB_Z3}
 
-z3: lib/z3/build/z3-dist
+clean-deps:
+	rm -rf lib/c4/build
+	rm -rf lib/z3/build
 
-lib/z3/build/z3-dist: lib/z3/build/libz3.dylib
-	# We need to make these parent directories so that Z3's Makefile
-	# doesn't complain about them missing when it copies files during
-	# the `install` step:
-	mkdir -p lib/z3/build/z3-dist/lib/python2.7/dist-packages
-	mkdir -p lib/z3/build/z3-dist/lib/python2.6/dist-packages
-	cd lib/z3/build && make install
+lib: .gitmodules
```

Comment:
Based on the CircleCI output, I think this is implicitly being called as part of `make deps`: https://circleci.com/gh/JoshRosen/molly/32

I think you're right and we should explicitly have `${LIB_Z3}` and `${LIB_C4}` depends on this, though.